components/EventCard/EventCard.component.js

```diff
@@ -8,24 +8,77 @@ import { UserContext } from '../../context/UserContext';
 
 import Button from '../MainButton/MainButton.component';
 
+const axios = require('axios');
+
 const EventCard = (props) => {
 
     const [isModalVisible, setModalVisible] = useState(false);
-    const { userEvents, setUserEvents } = useContext(UserContext);
-    
+    const { userEvents, addUserEvent } = useContext(UserContext);
+    const { token } = useContext(UserContext);
+
+    const subscribeToEvent = async (eventID) => {
+		const url = 'http://10.0.2.2:9090/event/register/'+eventID;
+
+		const settings = {
+			headers: {
+				'Content-Type': 'application/json',
+                'Authorization': 'Bearer ' + token
+			},
+		};
+
+		const body = {
+		}
+
+		try {
+            let response = await axios.post(url, body, settings);
+            console.log(response.data);
+            Alert.alert(response.data.message);
+            addUserEvent({
+                created_at : props.created_at,
+                event_id: props.event_id,
+                event_name: props.name,
+                info: props.info,
+                date: props.end_date,
+                perks: props.perks,
+                theme: props.theme,
+            })
+		} catch (error) {
+			console.error(error);
+		}
+    };
+    const deleteEvent = async (eventID) => {
+		const url = 'http://10.0.2.2:9090/event/delete_event/'+eventID;
+
+		const settings = {
+			headers: {
+				'Content-Type': 'application/json',
+                'Authorization': 'Bearer ' + token
+			},
+		};
+
+		const body = {
+		}
+
+		try {
+            let response = await axios.delete(url, settings);
+            console.log(response.data);
+		} catch (error) {
+			console.error(error);
+		}
+	};
```

Comment:
You might also want to code split this into a separate file, so we don't need to copy the huge chunk of code each time and can easily just import it.